packages/jest-mock/src/index.js

```diff
@@ -600,6 +612,15 @@ class ModuleMockerClass {
       mock.prototype.constructor = mock;
     }
 
+    if (metadata.subtype === 'react.component') {
```

Comment:
making this into `jest-mock` itself seems hairy. Is it possible to somehow extend `jest-mock` with plugins? So `getSubtype` existed, but it's possible to plug in specific subtypes, like React. Then Jest could extend it, but not couple `jest-mock` to React